src/main/java/org/minions/devfund/lourdes/battleship/Ship.java

```diff
@@ -0,0 +1,292 @@
+        final int totalLine = 3;
+        int startRow = row > 0 ? row - 1 : row;
+        int startColumn = column > 0 ? column - 1 : column;
+        int endRow = horizontal ? row - 1 : row + length;
+        int endColumn = horizontal ? column + length : column;
+        int borderLimit = row == 0 || column == 0 ? 2 : totalLine;
+        boolean isFreePosition = false;
+        if (checkFreeSpaceForShip(row, column, horizontal, ocean)) {
+            if (horizontal) {
+                isFreePosition = checkHorizontalSide(startRow, startColumn, ocean, endColumn, borderLimit);
+            } else {
+                isFreePosition = checkVerticalSide(startRow, startColumn, ocean, endRow, borderLimit);
+            }
+        }
+        return isFreePosition;
+    }
+
+    /**
+     * Method that verify if all horizontal sides are free.
+     *
+     * @param row         row position.
+     * @param column      column position.
+     * @param ocean       Ocean object.
+     * @param columnLimit column limit.
+     * @param rowLimit    row limit.
+     * @return true if all horizontal sides are free, false otherwise.
+     */
+    private boolean checkHorizontalSide(int row, int column, final Ocean ocean, int columnLimit, int rowLimit) {
+        for (int i = row; i < row + rowLimit; i++) {
+            for (int j = column; j < columnLimit; j++) {
+                if (ocean.isOccupied(i, j)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Method that verify all vertical sides are free.
+     *
+     * @param row         row position.
+     * @param column      column position.
+     * @param ocean       Ocean object.
+     * @param rowLimit    row limit.
+     * @param columnLimit column limit.
+     * @return true if all vertical sides are free, false otherwise.
+     */
+    private boolean checkVerticalSide(int row, int column, final Ocean ocean, int rowLimit, int columnLimit) {
+        for (int i = column; i < column + columnLimit; i++) {
+            for (int j = row; j < rowLimit; j++) {
+                if (ocean.isOccupied(j, i)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Method that check if a given position is free to place a ship.
+     *
+     * @param row        row position.
+     * @param column     column position.
+     * @param horizontal ship orientation.
+     * @param ocean      Ocean object where the ship will be placed.
+     * @return True if the ship is able to be placed in the given position, false otherwise.
+     */
+    private boolean checkFreeSpaceForShip(int row, int column, boolean horizontal, final Ocean ocean) {
+        try {
+            for (int i = 0; i < length; i++) {
+                if (horizontal && ocean.isOccupied(row, column + i)) {
+                    return false;
+
+                } else if (ocean.isOccupied(row + i, column)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This branch's code block is the same as the block for the branch on line 117. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1871)